### PR TITLE
Use Tentacle for the GitHub API

### PR DESCRIPTION
**Pull request by mdiep**
1 thing to note: This changes the deployment target from 10.9 to 10.10. I think that's reasonable, given that using 10.9 for deployment is pretty infeasible.

### CODE REVIEW

**Comment by NachoSoto** [comment]
:clap:

File: Carthage.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -168,10 +169,11 @@
...
 		89E80E631C755059000F8DCB /* BuildArgumentsSpec.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = BuildArgumentsSpec.swift; sourceTree = "<group>"; };
-		89E80E651C755B3D000F8DCB /* GitHubSpec.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = GitHubSpec.swift; sourceTree = "<group>"; };
 		89E80E681C755C1B000F8DCB /* FakeURLSession.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = FakeURLSession.swift; sourceTree = "<group>"; };
 		98400F5A1BD24DC5008C5DDE /* carthage-bash-completion */ = {isa = PBXFileReference; lastKnownFileType = text; path = "carthage-bash-completion"; sourceTree = "<group>"; };
 		98400F5B1BD24DC5008C5DDE /* carthage-zsh-completion */ = {isa = PBXFileReference; lastKnownFileType = text; path = "carthage-zsh-completion"; sourceTree = "<group>"; };
+		BEB076641C8A1FD800ABD373 /* Curry.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Curry.framework; path = ../../Carthage/Checkouts/Curry/build/Debug/Curry.framework; sourceTree = "<group>"; };
```
Comment (NachoSoto) [inline Carthage.xcodeproj/project.pbxproj]: I think this path is... not right?

**Comment by mdiep** [comment]
This is ready for another look! :tada:

**Comment by ikesyo** [comment]
Maybe `Source/carthage/Components.plist` should be updated.

**Comment by ikesyo** [comment]
`import Argo` in `FrameworkExtensions.swift` and `GitHub.swift` can be removed.

**Comment by mdiep** [comment]
Good catches!

**Comment by mdiep** [comment]
_bump_

**Comment by ikesyo** [comment]
Looks good to me :+1: Thanks for the great work! @mdiep

**Pull request merged**